service/src/main/java/datawave/microservice/dictionary/DataDictionaryWebController.java

```diff
@@ -0,0 +1,275 @@
+@Controller
+@RequestMapping(path = "/data/v1", produces = {MediaType.TEXT_HTML_VALUE})
+@EnableConfigurationProperties(DataDictionaryProperties.class)
+public class DataDictionaryWebController<DESC extends DescriptionBase<DESC>,DICT extends DataDictionaryBase<DICT,META>,META extends MetadataFieldBase<META,DESC>,FIELD extends DictionaryFieldBase<FIELD,DESC>,FIELDS extends FieldsBase<FIELDS,FIELD,DESC>> {
+    
+    private static final String EMPTY_STR = "", SEP = ", ";
+    
+    private final DataDictionaryProperties dataDictionaryConfiguration;
+    private final DataDictionary<META,DESC,FIELD> dataDictionary;
+    private final ResponseObjectFactory<DESC,DICT,META,FIELD,FIELDS> responseObjectFactory;
+    private final AccumuloConnectionService accumuloConnectionService;
+    
+    private final Consumer<META> TRANSFORM_EMPTY_INTERNAL_FIELD_NAMES = meta -> {
+        if (meta.getInternalFieldName() == null || meta.getInternalFieldName().isEmpty()) {
+            meta.setInternalFieldName(meta.getFieldName());
+        }
+    };
+    
+    public DataDictionaryWebController(DataDictionaryProperties dataDictionaryConfiguration, DataDictionary<META,DESC,FIELD> dataDictionary,
+                    ResponseObjectFactory<DESC,DICT,META,FIELD,FIELDS> responseObjectFactory, AccumuloConnectionService accumloConnectionService) {
+        this.dataDictionaryConfiguration = dataDictionaryConfiguration;
+        this.dataDictionary = dataDictionary;
+        this.responseObjectFactory = responseObjectFactory;
+        this.accumuloConnectionService = accumloConnectionService;
+        dataDictionary.setNormalizationMap(dataDictionaryConfiguration.getNormalizerMap());
+    }
+    
+    /**
+     * Returns the DataDictionary for the given parameters.
+     *
+     * @param modelName
+     *            Optional model name
+     * @param modelTableName
+     *            Optional model table name
+     * @param metadataTableName
+     *            Optional metadata table name
+     * @param queryAuthorizations
+     *            Optional query authorizations
+     * @param dataTypeFilters
+     *            Optional data type filters
+     * @param currentUser
+     *            the current user
+     * @return The ModelAndView for datadictionary.html which contains the data dictionary fields.
+     * @throws Exception
+     *             if there is any problem fetching the entries
+     */
+    @GetMapping("/")
+    @Timed(name = "dw.dictionary.data.get", absolute = true)
+    public ModelAndView get(@RequestParam(required = false) String modelName, @RequestParam(required = false) String modelTableName,
+                    @RequestParam(required = false) String metadataTableName, @RequestParam(name = "auths", required = false) String queryAuthorizations,
+                    @RequestParam(defaultValue = "") String dataTypeFilters, @AuthenticationPrincipal ProxiedUserDetails currentUser) throws Exception {
```

Comment:
FYI, another way to do MVC is to just return a string from the rest endpoint which represents the view name, and add a `Model` argument to the rest endpoint signature, which is where you would set all of your model fields.  Not asking you to change anything - just sharing info.

Reply:
Thank you for the info! Yeah, I read about doing it like that but I was afraid of changing the method signature haha